### PR TITLE
Add multi-tenancy support for signal broadcast in the java client

### LINKED ISSUE #13558: Java client supports multi-tenancy for BroadcastSignal RPC
### Description

The Java client provides a `BroadcastSignalCommand` for broadcasting tenant-aware signals in Zeebe. These commands should support multi-tenancy by exposing an optional `tenantId` property/method.

The following error codes may be returned:
* PERMISSION_DENIED (code: 7) 
   * when a user attempts to broadcast a signal of a tenant they are not authorized for, when multi-tenancy is enabled.
* INVALID_ARGUMENT (code: 3)
   * For a provided tenant id, when multi-tenancy is disabled
   * For a missing tenant id, when multi-tenancy is enabled
   * For an invalid tenant id (i.e. doesn't match the pre-defined format), when multi-tenancy is enabled.

### PR DESCRIPTION
**Pull request by koevskinikola**
## Description

Add multi-tenancy support for signal broadcast in the java client.

## Related issues

closes #13558 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by koevskinikola** [comment]
:information_source: I haven't added any IT tests with this PR, as the engine part of Signals hasn't been implemented yet, and deployment of tenant-aware processes that contain signals is currently disabled (except on the `<default>` tenant).

File: clients/java/revapi.json [hunk 1]
```diff
@@ -28,6 +28,13 @@
         {
           "justification": "Ignore new methods on all types, as these are not meant to be implemented but simply consumed; as such, new methods are perfectly fine to add",
           "code": "java.method.addedToInterface"
+        },
+        {
+          "ignore": true,
+          "code": "java.annotation.removed",
+          "old": "method java.lang.String io.camunda.zeebe.client.api.response.BroadcastSignalResponse::getTenantId()",
+          "new": "method java.lang.String io.camunda.zeebe.client.api.response.BroadcastSignalResponse::getTenantId()",
+          "justification": "The ExperimentalApi annotation was added to mark this method as 'in-development'. The functionality has since been implemented and the annotation isn't necessary."
```
Comment (remcowesterhoud) [inline clients/java/revapi.json:37]: 😄 I wonder if we can just say ignore any cases where the `@ExperimentalApi` was removed. It's basically made for removing.

Comment (remcowesterhoud) [reply clients/java/revapi.json:37]: I think this should work:

```
{
    "ignore": true,
    "code": "java.annotation.removed",
    "annotationType": "io.camunda.zeebe.client.api.ExperimentalApi",
    "justification": "The ExperimentalApi annotation is used to mark this methods as 'in-development'. It is okay to remove it after a feature is implemented"
}
```

**Review by remcowesterhoud — APPROVED** [review]
LGTM @koevskinikola 🚀 I have one suggestion regarding the revapi.

**Comment by koevskinikola** [comment]
bors merge

**Pull request merged**